src/Types.hs

```diff
@@ -88,38 +139,63 @@ data Category =
   deriving (Eq, Ord, Bounded, Enum, Show, Read, Generic)
 
 -- | "Verbesserungsvorschlag"
+--
+-- 'Comments' are hierarchical.  The application logic is responsible for putting some limit on the
+-- recursion depth under which all children become siblings.  (FIXME: or should we represent this in
+-- the types?)
```

Comment:
I think is better to simply make that restriction in the application logic it could then be easier to change.